src/test/java/com/puppycrawl/tools/checkstyle/checks/sizes/OuterTypeNumberCheckTest.java

```diff
@@ -104,4 +110,37 @@ public void testWithRecords() throws Exception {
                 getNonCompilablePath("InputOuterTypeNumberRecords.java"), expected);
     }
 
+    /**
+     * Checks if the private field {@code currentDepth} and {@code outerNum} is
+     * properly cleared during the start of processing the next file in the
+     * check as they are file specific values.
+     *
+     * @throws Exception if there is an error.
+     */
+    @Test
+    public void testClearState() throws Exception {
```

Comment:
I wonder... could https://github.com/checkstyle/checkstyle/blob/33c291e2dc1f846362902ed2906d84acb3777b74/src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java#L251 kill these mutations by sequential execution of check on files?

Reply:
No. It has to be the same check instance. BDD, like the verify methods, recreate the entire config each run.